src/constbuffer_array_splitter.c

```diff
@@ -208,3 +208,150 @@ CONSTBUFFER_ARRAY_HANDLE constbuffer_array_splitter_split(CONSTBUFFER_ARRAY_HAND
+                            {
+                                CONSTBUFFER_ARRAY_HANDLE arr = constbuffer_array_create_from_start_and_end(buffers, start_buffer_index, current_buffer_count, start_buffer_offset, buffer->size);
+                                if(arr == NULL)
+                                {
+                                    LogError("constbuffer_array_create_from_start_and_end failed buffers=%p, uint32_t start_buffer_index=%" PRIu32 ", uint32_t buffer_count=%" PRIu32 ", uint32_t start_buffer_offset=%" PRIu32 ", uint32_t end_buffer_size=%" PRIu32 ,
+                                        buffers, start_buffer_index, current_buffer_count, start_buffer_offset, buffer->size);
+                                    result = NULL;
+                                    break;
+                                }
+                                else
+                                {
+                                    result[index] = arr;
+                                }
+                            }
+                            i++;
+                        }
+                        else if(current_buffer_size + buffer->size > max_buffer_size)
+                        {
+                            current_buffer_count++;
+                            end_buffer_size = max_buffer_size - current_buffer_size;
+                            CONSTBUFFER_ARRAY_HANDLE arr = constbuffer_array_create_from_start_and_end(buffers, start_buffer_index, current_buffer_count, start_buffer_offset, end_buffer_size);
+                            if(arr == NULL)
+                            {
+                                LogError("constbuffer_array_create_from_start_and_end failed buffers=%p, uint32_t start_buffer_index=%" PRIu32 ", uint32_t buffer_count=%" PRIu32 ", uint32_t start_buffer_offset=%" PRIu32 ", uint32_t end_buffer_size=%" PRIu32 ,
+                                    buffers, start_buffer_index, current_buffer_count, start_buffer_offset, end_buffer_size);
+                                result = NULL;
+                                break;
+                            }
+                            else
+                            {
+                                result[index] = arr;
+                                current_buffer_size += end_buffer_size;
+                                total_size += current_buffer_size;
+
+                                //reset for another sub array
+                                start_buffer_index = i;
+                                start_buffer_offset = end_buffer_size;
+                                end_buffer_size = 0;
+                                current_buffer_count = 1;
+                                current_buffer_size = buffer->size - end_buffer_size;
+
+                                index++;
+                                i++;
+                            }
+                        }
+                        else
+                        {
```

Comment:
suspect that this can be combined with the case above (if greater than or equal). Loop will exit anyway in the "equal" case, but should be less code duplication.